src/scope.rs

```diff
@@ -57,27 +57,34 @@ enum Found {
 }
 
 /// Determines whether function usage matches function definitions letter-case wise
-type IsValidCasing = bool;
+enum Casing {
+    /// The same as in definition
+    Original,
+    /// Letter-casing differs from definition
+    Different,
+}
 
 impl<'a> Scope<'a> {
-    fn search(&self, name: &str) -> (Option<Found>, IsValidCasing) {
+    fn search(&self, name: &str) -> Option<(Found, Casing)> {
         let case_insensitive_name = UniCase::new(name);
-        match self.directly_imported_or_defined.get(&case_insensitive_name) {
-            None => {
-                match self.all.get(&case_insensitive_name) {
-                    None => {
-                        (None, true)
-                    }
-                    Some(val) => {
-                        let is_valid_casing = val.to_string() == name;
-                        (Some(Found::Indirect), is_valid_casing)
-                    }
-                }
-            }
-            Some(val) => {
-                let is_valid_casing = val.to_string() == name;
-                (Some(Found::Direct), is_valid_casing)
+
+        match self.all.get(&case_insensitive_name) {
+            Some(original_name) => {
+                let directness = if self.directly_imported_or_defined.contains(&case_insensitive_name) {
+                    Found::Direct
+                } else {
+                    Found::Indirect
+                };
+
+                let casing = if name == &***original_name {
```

Comment:
????????????????????

Reply:
Wanted to explain upon that before, but I've felt you'll ask anyway :P
So, usually, Rust goes through all the layers of references and smart pointers unless it doesn't.

The first `*` dereferences the `&Unicase<>` (we have a reference to `Unicase`, because set's `get` returns only reference to the value, which makes sense) and leaves us with `Unicase<&str>`.

The second `*` should get us from `Unicase` to `&str` (because `Unicase` is "kind of" smart pointer over what it wraps. See [the chapter on deref coercions](https://doc.rust-lang.org/book/second-edition/ch15-02-deref.html).

The third `*` gets us from `&str` to `str`.

The `&` gets us from `str` back to `&str`. Wait... I'm not sure why I did it that way, just `**` should be enough.